src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -1308,6 +1309,8 @@ int wm_vuldet_send_agent_report(sqlite3 *db, OSHash *cve_table, agent_software *
                 }
             }
 
+            //Save the vulnerability in the agent database
+            wdb_agents_vuln_cve_insert(atoi(report->agent_id), report->software, report->version, report->arch, report->cve, NULL);
```

Comment:
Vuln_det already has an open socket with WazuhDB. I think we should replace the NULL with **wm_vuldet_get_wdb_socket()** as a sock parameter, to re-use it.
If it isn't possible, we still can create a new sock and keep it open during all the **inserts** of the loop.

Reply:
You are right. Fixed!